api/handlers.go

```diff
@@ -1096,11 +1105,13 @@ func (api *API) trackUploadPOST(u *database.User, w http.ResponseWriter, req *ht
 	// administrative details, such as user's quotas check.
 	// Note that this call is not affected by the request's context, so we use
 	// a separate one.
-	go api.checkUserQuotas(context.Background(), u)
+	if u != nil && !u.ID.IsZero() {
+		go api.checkUserQuotas(context.Background(), u)
+	}
 }
 
 // trackDownloadPOST registers a new download in the system.
-func (api *API) trackDownloadPOST(u *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+func (api *API) trackDownloadPOST(_ *database.User, w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
```

Comment:
I wouldn't track anon downloads

Reply:
This will cause issues for sure, it's a huge amount of db records.  
I don't think it's a good idea to track anon downloads for that reason.

Reply:
I thought about this a bit more and I think we'll be adding about 70-80mb in storage space for every 1M downloads. So let's say we have 10M downloads a month, but probably it's closer to 20M and soon it'll be more, so every month we'll be easily adding 1GiB of storage. 

Then the second thing is that there's 3 indices, which is about 25mib per 1M documents. So again here after 1M of conservative downloads that's 250mib in extra index size, every month. Mongo usually keeps it entire index in RAM, not always and not all the time, but in general it does. When you have a mongo cluster you should generally make it so all indexes fit in memory. I'm not saying it won't, but it's not terribly scalable.

Then third is that I'm concerned about the writes to the PRIMARY, we write to a single primary and I have once experienced writes to easily take up to 200ms. Tracking anon downloads will essentially cause our total req/s to be writes/s to our PRIMARY. This could very well lead to worse write times.

I would love download stats, like everyone, I'm just not sure mongo is the place for it, or the tool to get that job done.
And yes, we can tackle this problem when it arises, I'm just saying that it will, and it will probably be in 3-6 maybe even 9 months from now, but it will cause issues so I don't see why we should add something we know we'll have to eventually fix.

Reply:
It's probably a ton of data but my impression from the description was that this will be removed in a f/u?

Reply:
Removed.